app/code/Magento/Ui/view/base/web/js/form/element/ui-select.js

```diff
@@ -1167,7 +1167,7 @@ define([
                 return;
             }
 
-            if (searchKey !== this.lastSearchKey) {
+            if (searchKey !== this.lastSearchKey || currentPage === 1) {
```

Comment:
to be honest I think we can even simplify this condition
```suggestion
            if (currentPage === 1) {
```